alpa/pipeline_parallel/cross_mesh_resharding.py

```diff
@@ -895,6 +943,46 @@ def __init__(self, per_spec_plans, is_local_allgather):
         self.is_local_allgather = is_local_allgather
 
 
+class SendRecvReshardingStrategyWithOrder(ReshardingStrategy):
```

Comment:
can we simply add the `order` attribute in `ReshardingStrategy` of which the default value is what we do in `_compile_send_recv_tasks` or `_compile_broadcast_tasks`:
```
            for i, (dst_tile, src_tiles,
                    _) in enumerate(self.task_spec.dst_tile_to_src_tiles_map):
                for k, _ in enumerate(dst_tile.replica_device_strs):
                    for j, _ in enumerate(src_tiles):
                        order.append((i, k, j))
```
The benefit is to remove the `hasattr` branch in `_compile_xxx_tasks` and `xxxReshardingStrategyWithOrder` classes.

Reply:
OK